torch/_inductor/codegen/simd.py

```diff
@@ -2034,9 +2034,37 @@ def convert_tiling_to_3d(
 
         if len(ranked_tilings) > 1:
             perf_hint_log.info("possibly bad tiling: %s", ranked_tilings)
+        # save iter vars in indirect broadcast memory reads
+        indirect_broadcast_pairs: list[tuple[sympy.Symbol, sympy.Symbol]] = []
+        if len(node_schedule) == 1:
+            for read in node_schedule[0].read_writes.reads:
+                if (
+                    isinstance(read, MemoryDep)
+                    and hasattr(read, "indirect_broadcast")
+                    and read.indirect_broadcast
+                ):
+                    indirect_broadcast_pairs.append(read.indirect_broadcast)
+
+            if indirect_broadcast_pairs:
```

Comment:
Add comment how this part works especially how you decide the new loop order

Reply:
added in https://github.com/pytorch/pytorch/pull/145992/commits/59e131647fe1bdc9a3c53b8643421629a44ff8fc